subsys/net/ip/net_pkt.c

```diff
@@ -1809,6 +1809,13 @@ static void clone_pkt_attributes(struct net_pkt *pkt, struct net_pkt *clone_pkt)
 	net_pkt_set_priority(clone_pkt, net_pkt_priority(pkt));
 	net_pkt_set_orig_iface(clone_pkt, net_pkt_orig_iface(pkt));
 	net_pkt_set_captured(clone_pkt, net_pkt_is_captured(pkt));
+	net_pkt_set_sent(clone_pkt, net_pkt_sent(pkt));
+	net_pkt_set_queued(clone_pkt, net_pkt_queued(pkt));
```

Comment:
Is this correct to clone those flags? If the original packet was marked as sent/queued, it does not seem to be true for the clone.

Anyway, both these flags seem no longer in use (they aren't read anywhere within the codebase). The setters are called in a couple of places, but according to the comments/git messages they seemed to have been used by the old TCP implementation (some comments even refer to obsolete TCP code).

Maybe if we're doing cleanup in the flags, we should consider removing them instead?

Reply:
> Is this correct to clone those flags? If the original packet was marked as sent/queued, it does not seem to be true for the clone.

I was thinking that too, but it is difficult to know what flags would not need to be cloned. The cloning implies also that we get identical copy of the original net_pkt so with in mind I considered it ok if we just copy all the flags.

> 
> Anyway, both these flags seem no longer in use (they aren't read anywhere within the codebase). The setters are called in a couple of places, but according to the comments/git messages they seemed to have been used by the old TCP implementation (some comments even refer to obsolete TCP code).

The sent flag is shared with eof, so while we could remove the function, the actual bit field cannot be removed (but could be renamed).

> 
> Maybe if we're doing cleanup in the flags, we should consider removing them instead?

Yes, that is good idea.

Reply:
> 

> Maybe if we're doing cleanup in the flags, we should consider removing them instead?

> Yes, that is good idea.

If this flag is really unused then maybe should be renamed to `unused` for future use.

Reply:
I removed all deprecated and unused flags together with setters/getters.